diff -ruN abseil-cpp-20240722.0-yb-1/MODULE.bazel abseil-cpp-20240722.0-yb-1-modified/MODULE.bazel
--- abseil-cpp-20240722.0-yb-1/MODULE.bazel
+++ abseil-cpp-20240722.0-yb-1-modified/MODULE.bazel
@@ -40,3 +40,9 @@
 
 bazel_dep(name = "platforms",
           version = "0.0.10")
+
+bazel_dep(name = "rules_cc",
+          version = "0.2.14")
+
+bazel_dep(name = "rules_foreign_cc",
+          version = "0.13.0")
diff -ruN abseil-cpp-20240722.0-yb-1/absl/BUILD.bazel abseil-cpp-20240722.0-yb-1-modified/absl/BUILD.bazel
--- abseil-cpp-20240722.0-yb-1/absl/BUILD.bazel
+++ abseil-cpp-20240722.0-yb-1-modified/absl/BUILD.bazel
@@ -21,6 +21,7 @@
     "ABSL_DEFAULT_LINKOPTS",
 )
 load("//absl:cc_static_library.bzl", "cc_static_library")
+load("@rules_cc//cc:defs.bzl", "cc_binary", "cc_library")
 
 package(default_visibility = ["//visibility:public"])
 
diff -ruN abseil-cpp-20240722.0-yb-1/absl/algorithm/BUILD.bazel abseil-cpp-20240722.0-yb-1-modified/absl/algorithm/BUILD.bazel
--- abseil-cpp-20240722.0-yb-1/absl/algorithm/BUILD.bazel
+++ abseil-cpp-20240722.0-yb-1-modified/absl/algorithm/BUILD.bazel
@@ -20,6 +20,7 @@
     "ABSL_DEFAULT_LINKOPTS",
     "ABSL_TEST_COPTS",
 )
+load("@rules_cc//cc:defs.bzl", "cc_library", "cc_test")
 
 package(
     default_visibility = ["//visibility:public"],
diff -ruN abseil-cpp-20240722.0-yb-1/absl/base/BUILD.bazel abseil-cpp-20240722.0-yb-1-modified/absl/base/BUILD.bazel
--- abseil-cpp-20240722.0-yb-1/absl/base/BUILD.bazel
+++ abseil-cpp-20240722.0-yb-1-modified/absl/base/BUILD.bazel
@@ -20,6 +20,7 @@
     "ABSL_DEFAULT_LINKOPTS",
     "ABSL_TEST_COPTS",
 )
+load("@rules_cc//cc:defs.bzl", "cc_binary", "cc_library", "cc_test")
 
 package(
     default_visibility = ["//visibility:public"],
diff -ruN abseil-cpp-20240722.0-yb-1/absl/cc_static_library.bzl abseil-cpp-20240722.0-yb-1-modified/absl/cc_static_library.bzl
--- abseil-cpp-20240722.0-yb-1/absl/cc_static_library.bzl
+++ abseil-cpp-20240722.0-yb-1-modified/absl/cc_static_library.bzl
@@ -1,12 +1,24 @@
+load("@rules_cc//cc/common:cc_info.bzl", "CcInfo")
+
 def _cc_static_library_impl(ctx):
     cc = ctx.attr.dep[CcInfo]
-    libraries = []
+    objects = []
     for link_input in cc.linking_context.linker_inputs.to_list():
         for library in link_input.libraries:
-            libraries += library.pic_objects
-    args = ["r", ctx.outputs.out.path] + [f.path for f in libraries]
+            if library.pic_objects:
+                objects += library.pic_objects
+            elif library.objects:
+                objects += library.objects
+    if not objects:
+        for link_input in cc.linking_context.linker_inputs.to_list():
+            for library in link_input.libraries:
+                if library.pic_static_library:
+                    objects.append(library.pic_static_library)
+                elif library.static_library:
+                    objects.append(library.static_library)
+    args = ["rcs", ctx.outputs.out.path] + [f.path for f in objects]
     ctx.actions.run(
-        inputs = libraries,
+        inputs = objects,
         outputs = [ctx.outputs.out],
         executable = "/usr/bin/ar",
         arguments = args,
diff -ruN abseil-cpp-20240722.0-yb-1/absl/cleanup/BUILD.bazel abseil-cpp-20240722.0-yb-1-modified/absl/cleanup/BUILD.bazel
--- abseil-cpp-20240722.0-yb-1/absl/cleanup/BUILD.bazel
+++ abseil-cpp-20240722.0-yb-1-modified/absl/cleanup/BUILD.bazel
@@ -18,6 +18,7 @@
     "ABSL_DEFAULT_LINKOPTS",
     "ABSL_TEST_COPTS",
 )
+load("@rules_cc//cc:defs.bzl", "cc_library", "cc_test")
 
 package(
     default_visibility = ["//visibility:public"],
diff -ruN abseil-cpp-20240722.0-yb-1/absl/container/BUILD.bazel abseil-cpp-20240722.0-yb-1-modified/absl/container/BUILD.bazel
--- abseil-cpp-20240722.0-yb-1/absl/container/BUILD.bazel
+++ abseil-cpp-20240722.0-yb-1-modified/absl/container/BUILD.bazel
@@ -20,6 +20,7 @@
     "ABSL_DEFAULT_LINKOPTS",
     "ABSL_TEST_COPTS",
 )
+load("@rules_cc//cc:defs.bzl", "cc_binary", "cc_library", "cc_test")
 
 package(
     default_visibility = ["//visibility:public"],
diff -ruN abseil-cpp-20240722.0-yb-1/absl/crc/BUILD.bazel abseil-cpp-20240722.0-yb-1-modified/absl/crc/BUILD.bazel
--- abseil-cpp-20240722.0-yb-1/absl/crc/BUILD.bazel
+++ abseil-cpp-20240722.0-yb-1-modified/absl/crc/BUILD.bazel
@@ -18,6 +18,7 @@
     "ABSL_DEFAULT_LINKOPTS",
     "ABSL_TEST_COPTS",
 )
+load("@rules_cc//cc:defs.bzl", "cc_binary", "cc_library", "cc_test")
 
 package(
     default_visibility = ["//visibility:private"],
diff -ruN abseil-cpp-20240722.0-yb-1/absl/debugging/BUILD.bazel abseil-cpp-20240722.0-yb-1-modified/absl/debugging/BUILD.bazel
--- abseil-cpp-20240722.0-yb-1/absl/debugging/BUILD.bazel
+++ abseil-cpp-20240722.0-yb-1-modified/absl/debugging/BUILD.bazel
@@ -20,6 +20,7 @@
     "ABSL_DEFAULT_LINKOPTS",
     "ABSL_TEST_COPTS",
 )
+load("@rules_cc//cc:defs.bzl", "cc_binary", "cc_library", "cc_test")
 
 package(
     default_visibility = ["//visibility:public"],
diff -ruN abseil-cpp-20240722.0-yb-1/absl/flags/BUILD.bazel abseil-cpp-20240722.0-yb-1-modified/absl/flags/BUILD.bazel
--- abseil-cpp-20240722.0-yb-1/absl/flags/BUILD.bazel
+++ abseil-cpp-20240722.0-yb-1-modified/absl/flags/BUILD.bazel
@@ -20,6 +20,7 @@
     "ABSL_DEFAULT_LINKOPTS",
     "ABSL_TEST_COPTS",
 )
+load("@rules_cc//cc:defs.bzl", "cc_binary", "cc_library", "cc_test")
 
 package(
     default_visibility = ["//visibility:public"],
diff -ruN abseil-cpp-20240722.0-yb-1/absl/functional/BUILD.bazel abseil-cpp-20240722.0-yb-1-modified/absl/functional/BUILD.bazel
--- abseil-cpp-20240722.0-yb-1/absl/functional/BUILD.bazel
+++ abseil-cpp-20240722.0-yb-1-modified/absl/functional/BUILD.bazel
@@ -20,6 +20,7 @@
     "ABSL_DEFAULT_LINKOPTS",
     "ABSL_TEST_COPTS",
 )
+load("@rules_cc//cc:defs.bzl", "cc_library", "cc_test")
 
 package(
     default_visibility = ["//visibility:public"],
diff -ruN abseil-cpp-20240722.0-yb-1/absl/hash/BUILD.bazel abseil-cpp-20240722.0-yb-1-modified/absl/hash/BUILD.bazel
--- abseil-cpp-20240722.0-yb-1/absl/hash/BUILD.bazel
+++ abseil-cpp-20240722.0-yb-1-modified/absl/hash/BUILD.bazel
@@ -20,6 +20,7 @@
     "ABSL_DEFAULT_LINKOPTS",
     "ABSL_TEST_COPTS",
 )
+load("@rules_cc//cc:defs.bzl", "cc_binary", "cc_library", "cc_test")
 
 package(
     default_visibility = ["//visibility:public"],
diff -ruN abseil-cpp-20240722.0-yb-1/absl/log/BUILD.bazel abseil-cpp-20240722.0-yb-1-modified/absl/log/BUILD.bazel
--- abseil-cpp-20240722.0-yb-1/absl/log/BUILD.bazel
+++ abseil-cpp-20240722.0-yb-1-modified/absl/log/BUILD.bazel
@@ -20,6 +20,7 @@
     "ABSL_DEFAULT_LINKOPTS",
     "ABSL_TEST_COPTS",
 )
+load("@rules_cc//cc:defs.bzl", "cc_library", "cc_test")
 
 package(
     default_visibility = ["//visibility:public"],
diff -ruN abseil-cpp-20240722.0-yb-1/absl/log/internal/BUILD.bazel abseil-cpp-20240722.0-yb-1-modified/absl/log/internal/BUILD.bazel
--- abseil-cpp-20240722.0-yb-1/absl/log/internal/BUILD.bazel
+++ abseil-cpp-20240722.0-yb-1-modified/absl/log/internal/BUILD.bazel
@@ -20,6 +20,7 @@
     "ABSL_DEFAULT_LINKOPTS",
     "ABSL_TEST_COPTS",
 )
+load("@rules_cc//cc:defs.bzl", "cc_binary", "cc_library", "cc_test")
 
 package(
     default_visibility = [
diff -ruN abseil-cpp-20240722.0-yb-1/absl/memory/BUILD.bazel abseil-cpp-20240722.0-yb-1-modified/absl/memory/BUILD.bazel
--- abseil-cpp-20240722.0-yb-1/absl/memory/BUILD.bazel
+++ abseil-cpp-20240722.0-yb-1-modified/absl/memory/BUILD.bazel
@@ -20,6 +20,7 @@
     "ABSL_DEFAULT_LINKOPTS",
     "ABSL_TEST_COPTS",
 )
+load("@rules_cc//cc:defs.bzl", "cc_library", "cc_test")
 
 package(
     default_visibility = ["//visibility:public"],
diff -ruN abseil-cpp-20240722.0-yb-1/absl/meta/BUILD.bazel abseil-cpp-20240722.0-yb-1-modified/absl/meta/BUILD.bazel
--- abseil-cpp-20240722.0-yb-1/absl/meta/BUILD.bazel
+++ abseil-cpp-20240722.0-yb-1-modified/absl/meta/BUILD.bazel
@@ -20,6 +20,7 @@
     "ABSL_DEFAULT_LINKOPTS",
     "ABSL_TEST_COPTS",
 )
+load("@rules_cc//cc:defs.bzl", "cc_library", "cc_test")
 
 package(
     default_visibility = ["//visibility:public"],
diff -ruN abseil-cpp-20240722.0-yb-1/absl/numeric/BUILD.bazel abseil-cpp-20240722.0-yb-1-modified/absl/numeric/BUILD.bazel
--- abseil-cpp-20240722.0-yb-1/absl/numeric/BUILD.bazel
+++ abseil-cpp-20240722.0-yb-1-modified/absl/numeric/BUILD.bazel
@@ -18,6 +18,7 @@
     "ABSL_DEFAULT_LINKOPTS",
     "ABSL_TEST_COPTS",
 )
+load("@rules_cc//cc:defs.bzl", "cc_binary", "cc_library", "cc_test")
 
 package(
     default_visibility = ["//visibility:public"],
diff -ruN abseil-cpp-20240722.0-yb-1/absl/profiling/BUILD.bazel abseil-cpp-20240722.0-yb-1-modified/absl/profiling/BUILD.bazel
--- abseil-cpp-20240722.0-yb-1/absl/profiling/BUILD.bazel
+++ abseil-cpp-20240722.0-yb-1-modified/absl/profiling/BUILD.bazel
@@ -18,6 +18,7 @@
     "ABSL_DEFAULT_LINKOPTS",
     "ABSL_TEST_COPTS",
 )
+load("@rules_cc//cc:defs.bzl", "cc_binary", "cc_library", "cc_test")
 
 package(
     default_visibility = ["//visibility:private"],
diff -ruN abseil-cpp-20240722.0-yb-1/absl/random/BUILD.bazel abseil-cpp-20240722.0-yb-1-modified/absl/random/BUILD.bazel
--- abseil-cpp-20240722.0-yb-1/absl/random/BUILD.bazel
+++ abseil-cpp-20240722.0-yb-1-modified/absl/random/BUILD.bazel
@@ -22,6 +22,7 @@
     "ABSL_DEFAULT_LINKOPTS",
     "ABSL_TEST_COPTS",
 )
+load("@rules_cc//cc:defs.bzl", "cc_binary", "cc_library", "cc_test")
 
 package(
     default_visibility = ["//visibility:public"],
diff -ruN abseil-cpp-20240722.0-yb-1/absl/random/internal/BUILD.bazel abseil-cpp-20240722.0-yb-1-modified/absl/random/internal/BUILD.bazel
--- abseil-cpp-20240722.0-yb-1/absl/random/internal/BUILD.bazel
+++ abseil-cpp-20240722.0-yb-1-modified/absl/random/internal/BUILD.bazel
@@ -23,6 +23,7 @@
     "ABSL_TEST_COPTS",
     "absl_random_randen_copts_init",
 )
+load("@rules_cc//cc:defs.bzl", "cc_binary", "cc_library", "cc_test")
 
 default_package_visibility = [
     "//absl/random:__pkg__",
diff -ruN abseil-cpp-20240722.0-yb-1/absl/status/BUILD.bazel abseil-cpp-20240722.0-yb-1-modified/absl/status/BUILD.bazel
--- abseil-cpp-20240722.0-yb-1/absl/status/BUILD.bazel
+++ abseil-cpp-20240722.0-yb-1-modified/absl/status/BUILD.bazel
@@ -23,6 +23,7 @@
     "ABSL_DEFAULT_LINKOPTS",
     "ABSL_TEST_COPTS",
 )
+load("@rules_cc//cc:defs.bzl", "cc_library", "cc_test")
 
 package(
     default_visibility = ["//visibility:public"],
diff -ruN abseil-cpp-20240722.0-yb-1/absl/strings/BUILD.bazel abseil-cpp-20240722.0-yb-1-modified/absl/strings/BUILD.bazel
--- abseil-cpp-20240722.0-yb-1/absl/strings/BUILD.bazel
+++ abseil-cpp-20240722.0-yb-1-modified/absl/strings/BUILD.bazel
@@ -19,6 +19,7 @@
     "ABSL_DEFAULT_LINKOPTS",
     "ABSL_TEST_COPTS",
 )
+load("@rules_cc//cc:defs.bzl", "cc_binary", "cc_library", "cc_test")
 
 package(
     default_visibility = ["//visibility:public"],
diff -ruN abseil-cpp-20240722.0-yb-1/absl/synchronization/BUILD.bazel abseil-cpp-20240722.0-yb-1-modified/absl/synchronization/BUILD.bazel
--- abseil-cpp-20240722.0-yb-1/absl/synchronization/BUILD.bazel
+++ abseil-cpp-20240722.0-yb-1-modified/absl/synchronization/BUILD.bazel
@@ -20,6 +20,7 @@
     "ABSL_DEFAULT_LINKOPTS",
     "ABSL_TEST_COPTS",
 )
+load("@rules_cc//cc:defs.bzl", "cc_binary", "cc_library", "cc_test")
 
 package(
     default_visibility = ["//visibility:private"],
diff -ruN abseil-cpp-20240722.0-yb-1/absl/time/BUILD.bazel abseil-cpp-20240722.0-yb-1-modified/absl/time/BUILD.bazel
--- abseil-cpp-20240722.0-yb-1/absl/time/BUILD.bazel
+++ abseil-cpp-20240722.0-yb-1-modified/absl/time/BUILD.bazel
@@ -20,6 +20,7 @@
     "ABSL_DEFAULT_LINKOPTS",
     "ABSL_TEST_COPTS",
 )
+load("@rules_cc//cc:defs.bzl", "cc_library", "cc_test")
 
 package(
     default_visibility = ["//visibility:public"],
diff -ruN abseil-cpp-20240722.0-yb-1/absl/time/internal/cctz/BUILD.bazel abseil-cpp-20240722.0-yb-1-modified/absl/time/internal/cctz/BUILD.bazel
--- abseil-cpp-20240722.0-yb-1/absl/time/internal/cctz/BUILD.bazel
+++ abseil-cpp-20240722.0-yb-1-modified/absl/time/internal/cctz/BUILD.bazel
@@ -12,6 +12,7 @@
 #   See the License for the specific language governing permissions and
 #   limitations under the License.
 
+load("@rules_cc//cc:defs.bzl", "cc_library", "cc_test")
 package(features = [
     "header_modules",
     "layering_check",
diff -ruN abseil-cpp-20240722.0-yb-1/absl/types/BUILD.bazel abseil-cpp-20240722.0-yb-1-modified/absl/types/BUILD.bazel
--- abseil-cpp-20240722.0-yb-1/absl/types/BUILD.bazel
+++ abseil-cpp-20240722.0-yb-1-modified/absl/types/BUILD.bazel
@@ -19,6 +19,7 @@
     "ABSL_DEFAULT_LINKOPTS",
     "ABSL_TEST_COPTS",
 )
+load("@rules_cc//cc:defs.bzl", "cc_library", "cc_test")
 
 package(
     default_visibility = ["//visibility:public"],
diff -ruN abseil-cpp-20240722.0-yb-1/absl/utility/BUILD.bazel abseil-cpp-20240722.0-yb-1-modified/absl/utility/BUILD.bazel
--- abseil-cpp-20240722.0-yb-1/absl/utility/BUILD.bazel
+++ abseil-cpp-20240722.0-yb-1-modified/absl/utility/BUILD.bazel
@@ -20,6 +20,7 @@
     "ABSL_DEFAULT_LINKOPTS",
     "ABSL_TEST_COPTS",
 )
+load("@rules_cc//cc:defs.bzl", "cc_library", "cc_test")
 
 package(
     default_visibility = ["//visibility:public"],
