diff -au -ruN a/LIBRARY/makefile b/LIBRARY/makefile
--- a/LIBRARY/makefile	2025-04-03 02:21:32.349164560 +0000
+++ b/LIBRARY/makefile	2025-04-03 00:26:03.483892911 +0000
@@ -174,6 +174,8 @@
 # get rid of extra blank characters
 CFLAGS        := $(foreach n,$(CFLAGS),$n)
 
+CFLAGS := $(filter-out -Wno-error=unused-command-line-argument, $(CFLAGS))
+
 #========================================================================
 #   Added BID build options (used for open source release)
 #========================================================================
diff -au -ruN a/LIBRARY/float128/dpml_exception.c b/LIBRARY/float128/dpml_exception.c
--- a/LIBRARY/float128/dpml_exception.c	2025-03-23 14:39:16
+++ b/LIBRARY/float128/dpml_exception.c	2025-03-23 14:37:46
@@ -132,6 +132,6 @@
     !defined(wnt)
 
-#   include <sys/signal.h>
+#   include <signal.h>
 #   define DPML_SIGNAL(p)	 raise(SIGFPE)
 
 #else
