diff -ruN a/LIBRARY/src/makefile.iml_head b/LIBRARY/src/makefile.iml_head
--- a/LIBRARY/makefile.iml_head    2018-06-07 23:09:37.000000000 +0000
+++ a/LIBRARY/makefile.iml_head    2025-04-03 02:27:17.047879100 +0000
@@ -522,7 +522,7 @@
     endif
 endif
 
-CC_NAME_LIST  := icc icl gcc cl cc
+CC_NAME_LIST  := icc icl gcc cl cc compiler_wrapper_cc.py
 CC_TYPE_LIST  := gcc cl  gcc cl gcc
 CC_TYPES      := gcc cl
 INTEL_CC_LIST := icc icl
diff -ruN a/LIBRARY/src/bid128_pow.c b/LIBRARY/src/bid128_pow.c
--- a/LIBRARY/src/bid128_pow.c	2025-03-23 14:38:52
+++ b/LIBRARY/src/bid128_pow.c	2025-03-23 14:37:46
@@ -27,6 +27,7 @@
   THE POSSIBILITY OF SUCH DAMAGE.
 ******************************************************************************/

+#include <stdlib.h>
 #include "bid_trans.h"

 static BID_UINT128 BID128_0 = {BID128_LH_INIT( 0x0000000000000000ull, 0x3040000000000000ull )};
diff -ruN a/LIBRARY/src/bid32_pow.c b/LIBRARY/src/bid32_pow.c
--- a/LIBRARY/src/bid32_pow.c	2025-03-23 14:38:24
+++ b/LIBRARY/src/bid32_pow.c	2025-03-23 14:37:46
@@ -28,6 +28,7 @@
 ******************************************************************************/


+#include <stdlib.h>
 #include "bid_internal.h"

 #define BID32_NAN 0x7c000000ul
diff -ruN a/LIBRARY/src/bid64_pow.c b/LIBRARY/src/bid64_pow.c
--- a/LIBRARY/src/bid64_pow.c	2025-03-23 14:38:42
+++ b/LIBRARY/src/bid64_pow.c	2025-03-23 14:37:46
@@ -28,6 +28,7 @@
 ******************************************************************************/


+#include <stdlib.h>
 #include "bid_trans.h"

 #define BID64_0 0x31c0000000000000ull
