diff --git src/glog/vlog_is_on.h.in src/glog/vlog_is_on.h.in
index d325434..f138177 100644
--- src/glog/vlog_is_on.h.in
+++ src/glog/vlog_is_on.h.in
@@ -73,6 +73,21 @@
 #endif

 #if defined(__GNUC__)
+
+#ifdef THREAD_SANITIZER
+}
+extern "C" {
+void AnnotateIgnoreReadsBegin(const char *file, int line);
+void AnnotateIgnoreReadsEnd(const char *file, int line);
+}
+#define GLOG_IGNORE_READS_BEGIN() AnnotateIgnoreReadsBegin(__FILE__, __LINE__)
+#define GLOG_IGNORE_READS_END() AnnotateIgnoreReadsEnd(__FILE__, __LINE__)
+namespace google {
+#else
+#define GLOG_IGNORE_READS_BEGIN()
+#define GLOG_IGNORE_READS_END()
+#endif
+
 // We emit an anonymous static int* variable at every VLOG_IS_ON(n) site.
 // (Normally) the first time every VLOG_IS_ON(n) site is hit,
 // we determine what variable will dynamically control logging at this site:
@@ -83,8 +98,12 @@
   __extension__  \
   ({ static @ac_google_namespace@::SiteFlag vlocal__{NULL, NULL, 0, NULL};       \
      @ac_google_namespace@::int32 verbose_level__ = (verboselevel);                    \
+     GLOG_IGNORE_READS_BEGIN();                                                        \
+     bool result =                                                                     \
      (vlocal__.level == NULL ? @ac_google_namespace@::InitVLOG3__(&vlocal__, &FLAGS_v, \
                         __FILE__, verbose_level__) : *vlocal__.level >= verbose_level__); \
+     GLOG_IGNORE_READS_END();                                                          \
+     result;                                                                           \
   })
 #else
 // GNU extensions not available, so we do not support --vmodule.