diff --git a/src/symbolize.cc b/src/symbolize.cc
index 1ffc607..91422bd 100644
--- a/src/symbolize.cc
+++ b/src/symbolize.cc
@@ -843,7 +843,7 @@ static ATTRIBUTE_NOINLINE bool SymbolizeAndDemangle(void *pc, char *out,
                                                     int out_size) {
   Dl_info info;
   if (dladdr(pc, &info)) {
-    if ((int)strlen(info.dli_sname) < out_size) {
+    if (info.dli_sname && (int)strlen(info.dli_sname) < out_size) {
       strcpy(out, info.dli_sname);
       // Symbolization succeeded.  Now we try to demangle the symbol.
       DemangleInplace(out, out_size);
