diff -ruN a/LIBRARY/src/bid128_pow.c b/LIBRARY/src/bid128_pow.c
--- a/LIBRARY/src/bid128_pow.c	2025-03-23 14:38:52
+++ b/LIBRARY/src/bid128_pow.c	2025-03-23 14:37:46
@@ -27,6 +27,7 @@
   THE POSSIBILITY OF SUCH DAMAGE.
 ******************************************************************************/
 
+#include <stdlib.h>
 #include "bid_trans.h"
 
 static BID_UINT128 BID128_0 = {BID128_LH_INIT( 0x0000000000000000ull, 0x3040000000000000ull )};
diff -ruN a/LIBRARY/src/bid32_pow.c b/LIBRARY/src/bid32_pow.c
--- a/LIBRARY/src/bid32_pow.c	2025-03-23 14:38:24
+++ b/LIBRARY/src/bid32_pow.c	2025-03-23 14:37:46
@@ -28,6 +28,7 @@
 ******************************************************************************/
 
 
+#include <stdlib.h>
 #include "bid_internal.h"
 
 #define BID32_NAN 0x7c000000ul
diff -ruN a/LIBRARY/src/bid64_pow.c b/LIBRARY/src/bid64_pow.c
--- a/LIBRARY/src/bid64_pow.c	2025-03-23 14:38:42
+++ b/LIBRARY/src/bid64_pow.c	2025-03-23 14:37:46
@@ -28,6 +28,7 @@
 ******************************************************************************/
 
 
+#include <stdlib.h>
 #include "bid_trans.h"
 
 #define BID64_0 0x31c0000000000000ull
